File: Frontend/src/app/system/_domains/ProductEntry.ts
import { Injectable } from '@angular/core';
import { _EntityBase } from './_EntityBase'
import { Product } from './Product'
import { Entry } from './Entry';

@Injectable()
export class ProductEntry extends _EntityBase {
    public UnitPrice: number;
    public ICMS: number;
    public IPI: number;
    public Quantity: number;
    public IdProduct: number;
    public FKProduct: Product;
    public IdEntry: number;
    public FKEntry: Entry;

}

File: Backend/Stock.Domain/ProductEntry.cs
using Framework.Domain;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text;

namespace Stock.Domain
{
    public class ProductEntry : EntityBase
    {
        [Required(ErrorMessage = "O produto é obrigatório")]
        [ForeignKey(nameof(FKProduct))]
        public int IdProduct { get; set; }
        public Product FKProduct { get; set; }

        [ForeignKey(nameof(FKEntry))]
        public int IdEntry { get; set; }
        public Entry FKEntry { get; set; }

        [Required(ErrorMessage = "O preço unitário é obrigatório")]
        public decimal UnitPrice { get; set; }

        public decimal ICMS { get; set; }

        [Required(ErrorMessage = "O valor do IPI é obrigatório")]
        public decimal IPI { get; set; }

        [Required(ErrorMessage = "A quantidade é obrigatória")]
        public decimal Quantity { get; set; }
    }
}
